sync/src/block/extension.rs

```diff
@@ -717,6 +718,20 @@ impl Extension {
                     .block_header(&BlockId::Hash(hash))
                     .expect("Downloaded body's header must exist")
                     .decode();
+                let parent_transactions_root = self
+                    .client
+                    .block_header(&(*header.parent_hash()).into())
+                    .expect("The parent header must exist")
+                    .view()
+                    .transactions_root();
+                let calculated_transactions_root =
+                    skewed_merkle_root(parent_transactions_root, transactions.iter().map(Encodable::rlp_bytes));
+                if *header.transactions_root() != calculated_transactions_root {
+                    cwarn!(SYNC, "Received corrupted body for ${}({}", header.number(), hash);
+                    self.body_downloader.remove_target(&[hash]);
+                    continue
```

Comment:
I'm not sure whether calling `remove_target` and `continue` is safe or not. I'll investigate it.